iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/FileImplTest.java

```diff
@@ -19,113 +19,56 @@
  */
 package org.sonar.iac.docker.tree.impl;
 
-import org.junit.jupiter.api.Disabled;
+import com.sonar.sslr.api.RecognitionException;
 import org.junit.jupiter.api.Test;
 import org.sonar.iac.common.api.tree.impl.TextRanges;
 import org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar;
 import org.sonar.iac.docker.tree.api.DockerTree;
 import org.sonar.iac.docker.tree.api.File;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.sonar.iac.common.testing.TextRangeAssert.assertTextRange;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.sonar.iac.docker.tree.impl.DockerTestUtils.parse;
 
 class FileImplTest {
 
   @Test
-  void shouldParseEmptyFile() {
-    File file = parseFile("");
-    assertThat(file.getKind()).isEqualTo(DockerTree.Kind.FILE);
-    assertTextRange(file.textRange()).hasRange(1,0,1,0);
-    assertThat(file.globalArgs()).isEmpty();
-    assertThat(file.dockerImages()).isEmpty();
+  void shouldFailOnEmptyFile() {
+    assertThrows(RecognitionException.class, () -> parseFile(""), "Parse error at line 1 column 1");
   }
 
   @Test
-  void shouldParseEmptyFileWithByteOrderMark() {
-    File file = parseFile("\uFEFF");
-    assertThat(file.getKind()).isEqualTo(DockerTree.Kind.FILE);
-    assertTextRange(file.textRange()).hasRange(1,0,1,0);
-    assertThat(file.globalArgs()).isEmpty();
-    assertThat(file.dockerImages()).isEmpty();
-  }
```

Comment:
Please add another test case with a valid body and a byte order mark. So we can test if the parser works correctly.